Automattic/wp-calypso
# Newsletter Accent Color: Send site_accent_color with newsletter site create request

#### Proposed Changes

* when the accent color is selected in the `newsletter-signup` flow, the client is redirected to `signup`'s  `newsletter` flow (`/start/newsletter/domains`). Set `siteAccentColor` url arg if the accent color was selected.
* In the `domains` step obtain the `siteAccentColor` from the query args only if the current flow is `newsletter` and the arg was set. Otherwise do not include `siteAccentColor` in the step data object passed to the `submitSignupStep` function.
* Try to obtain the `siteAccentColor` from `stepData` in `createSiteWithCart` and if the value is truthy then include the value as `options.site_accent_color` property in the create new site request (`/sites/new`)

#### Testing Instructions



* Go to http://calypso.localhost:3000/setup/subscribers?flow=newsletter
* Follow the flow and select an accent color
* Ensure the accent color value you selected is included in the `.../sites/new` request's body as `options.site_accent_color`
![Screen Shot 2022-08-23 at 14 37 55](https://user-images.githubusercontent.com/2019970/186160371-fa0920cc-014c-4cde-8e8e-86eea30c1168.png)


#### Pre-merge Checklist






Resolves #66765
